aio_pika/robust_connection.py

```diff
@@ -106,15 +106,25 @@ def add_reconnect_callback(self, callback: Callable[[], None]):
 
         try:
             await super().connect()
-        except CONNECTION_EXCEPTIONS:
+
+            for number, channel in self._channels.items():
+                await channel.on_reconnect(self, number)
+
+            self._reconnect_callbacks(self)
+
+        except CONNECTION_EXCEPTIONS as e:
             log.exception('Connection attempt error')
 
+            if self.connection is not None:
+                await asyncio.gather(
+                    self.connection.close(e), return_exceptions=True
```

Comment:
If you use `return_exceptions=True` to avoid exceptions raised maybe should use following:
```python
with suppress(CONNECTION_EXCEPTIONS):
    await self.connection.close(e)
```

Reply:
gather creates a separate task from coroutine. I think this better.